lib/chef/provider/package/apt.rb

```diff
@@ -50,86 +56,91 @@ def default_release_options
+              else
+                Chef::Log.debug("#{@new_resource} candidate version is #{$1}")
               end
             end
-            if package.is_a?(Array)
-              final_installed_version << installed_version
-              final_candidate_version << candidate_version
-              final_installed << installed
-              final_virtual << virtual
-            else
-              final_installed_version = installed_version
-              final_candidate_version = candidate_version
-              final_installed = installed
-              final_virtual = virtual
-            end
           end
-          @candidate_version = final_candidate_version
-          @current_resource.version(final_installed_version)
-          @is_virtual_package = final_virtual
- 
-          return final_installed.is_a?(Array) ? final_installed.any? : final_installed
+
+          return {
+            installed_version:   installed_version,
+            installed:           installed,
+            candidate_version:   candidate_version,
+            is_virtual_package:  is_virtual_package,
+          }
         end
 
-        def install_package(name, version)
-          if name.is_a?(Array)
-            index = 0
-            package_name = name.zip(version).map do |x, y|
-              namestr = nil
-              if @is_virtual_package[index]
-                namestr = x
-              else
-                namestr = "#{x}=#{y}"
-              end
-              index += 1
-              namestr
-            end.join(' ')
+        def check_all_packages_state(package)
+          installed_version = {}
+          candidate_version = {}
+          installed = {}
+
+          [package].flatten.each do |pkg|
+            ret = check_package_state(pkg)
+            is_virtual_package[pkg] = ret[:is_virtual_package]
+            installed[pkg]          = ret[:installed]
+            installed_version[pkg]  = ret[:installed_version]
+            candidate_version[pkg]  = ret[:candidate_version]
+          end
+
+          if package.is_a?(Array)
+            @candidate_version = []
+            final_installed_version = []
+            [package].flatten.each do |pkg|
+              @candidate_version << candidate_version[pkg]
+              final_installed_version << installed_version[pkg]
+            end
+            @current_resource.version(final_installed_version)
           else
-            package_name = "#{name}=#{version}"
-            package_name = name if @is_virtual_package
+            @candidate_version = candidate_version[package]
+            @current_resource.version(installed_version[package])
           end
+        end
+
+        def install_package(name, version)
+          name_array = [ name ].flatten
+          version_array = [ version ].flatten
+          package_name = name_array.zip(version_array).map do |n, v|
+            is_virtual_package[n] ? n : "#{n}=#{v}"
```

Comment:
nope, its declared an accessor, better to drop the `@`

i was tempted to rage fix and drop all the `@` signs everywhere on accessors (all the `@new_resource` and `@current_resource` stuff makes my eyes bleed now), but want to keep this mostly confined to fixing the bug and not a style overhaul.